tst/EnergyPlus/unit/WaterThermalTanks.unit.cc

```diff
@@ -2053,3 +2053,232 @@ TEST_F(EnergyPlusFixture, StratifiedTankCalc)
 
 
 }
+
+TEST_F(EnergyPlusFixture, StratifiedTankDesuperheaterSourceHeat)
```

Comment:
Good unit test! A lot of these end up being setup. It might be helpful to have a few more comments about what the test is doing at the point it's doing the actual testing to make it a little easier to follow.

Reply:
Thanks for your comment, Noel. 
Sure, I will go through your comment and update soon. 😃